includes/class-wc-payments-order-service.php

```diff
@@ -1015,34 +1015,36 @@ private function mark_payment_authorized( $order, $intent_data ) {
 	/**
 	 * Updates an order to processing/completed status, while adding a note with a link to the transaction.
 	 *
-	 * @param WC_Order $order         Order object.
-	 * @param array    $intent_data   The intent data associated with this order.
+	 * @param WC_Order                          $order         Order object.
+	 * @param WC_Payments_API_Payment_Intention $intent        The intent instance.
 	 *
 	 * @return void
 	 */
-	private function mark_payment_capture_completed( $order, $intent_data ) {
-		$note = $this->generate_capture_success_note( $order, $intent_data['intent_id'], $intent_data['charge_id'] );
+	private function mark_payment_capture_completed( $order, $intent ) {
```

Comment:
The changes in this method are great; I really like how the OOP approach here improves readability and is more friendly to code analysis tools. Thanks!

Reply:
Agree, thanks